astropy/coordinates/representation.py

```diff
@@ -223,6 +223,24 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
```

Comment:
@StuartLittlefair - in @eteq's review of this in #5253, the addition of `__eq__` came up again, as it breaks backward compatibility. I just checked and everything runs fine if I simply remove this again, which leaves me even more confused. My sense right now would be to leave it out. If we do want equality checks, we can do that in a separate PR.